### PR TITLE
Move private secret key out of AttestationAuthority.

### PR DESCRIPTION
**Pull request by ooq**
This change is part of binauthz compatibility changes.
It will move private secret key name out of kritis attestation authority, and into the policies, i.e., the ISP policy and the build policy. 
The purpose of this is to separate the semantics of `signing` and `verifying`, leaving attestation authority to do only `verifying` and the respective policies to do `signing`.

### CODE REVIEW

File: pkg/kritis/apis/kritis/v1beta1/securitypolicy.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ type ImageSecurityPolicySpec struct {
 	ImageAllowlist                   []string                         `json:"imageAllowlist"`
 	PackageVulnerabilityRequirements PackageVulnerabilityRequirements `json:"packageVulnerabilityRequirements"`
 	AttestationAuthorityName         string                           `json:"attestationAuthorityName"`
```
Comment (nenaddedic) [inline pkg/kritis/apis/kritis/v1beta1/securitypolicy.go]: Let's make sure to also change this to Note name.  (Same in build policy.)  We can, and maybe should, do it in a separate PR, let's just not forget about it.

File: pkg/kritis/gcbsigner/signer.go [hunk 2]
```diff
@@ -66,14 +66,14 @@ func (s Signer) ValidateAndSign(prov BuildProvenance, bps []v1beta1.BuildPolicy)
...
-		if err := s.addAttestation(prov.ImageRef, bp.Namespace, bp.Spec.AttestationAuthorityName); err != nil {
+		if err := s.addAttestation(prov.ImageRef, bp.Namespace, bp.Spec.AttestationAuthorityName, bp.Spec.PrivateKeySecretName); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (s Signer) addAttestation(image string, ns string, authority string) error {
+func (s Signer) addAttestation(image string, ns string, authority string, keyName string) error {
```
Comment (nenaddedic) [inline pkg/kritis/gcbsigner/signer.go]: Maybe call it keySecretName or something to make it clear this is a k8s secret name, not key id.

File: integration/crd.go [hunk 3]
```diff
@@ -38,11 +38,10 @@ metadata:
   name: test-attestor
 spec:
   noteReference: projects/%s
-  privateKeySecretName: %s
   publicKeyData: %s`
 )
 
-// Secret name for test-attestor
+// Secret name for test-attestor, which matches integration/testdata/image-security-policy/my-isp.yaml
```
Comment (nenaddedic) [inline integration/crd.go]: Please consider moving the key pair generation out of createAA, because AA now only has the public portion.  One way to do it:
- move createAA call out of here and into setupISP() in integration/run_test.go
- rename this to createSecret or something like it
  + and change this to return the generated public key

WDYT?

File: pkg/kritis/apis/kritis/v1beta1/securitypolicy.go (hunk 1 above)
Comment (ooq) [reply pkg/kritis/apis/kritis/v1beta1/securitypolicy.go]: Ack.

File: integration/crd.go (hunk 3 above)
Comment (ooq) [reply integration/crd.go]: Done.

File: pkg/kritis/gcbsigner/signer.go (hunk 2 above)
Comment (ooq) [reply pkg/kritis/gcbsigner/signer.go]: Done.

File: integration/run_test.go [hunk 4]
```diff
@@ -278,7 +278,13 @@ func setUp(t *testing.T) (kubernetes.Interface, *v1.Namespace, func(t *testing.T
 		t.Fatalf("install: %v", err)
 	}
 
-	createAttestationAuthority(t, *gcpProject, ns.Name)
+	// Generate a key value pair
+	pubKey, privKey := testutil.CreateKeyPair(t, aaSecret)
```
Comment (nenaddedic) [inline integration/run_test.go]: I think 'aaSecret' can go.  Just write 'test-attestor' here and comment that it must be the same as in my-isp.yaml.

Comment (ooq) [reply integration/run_test.go]: It's also used in "cleanup.go". so it might still be easier to centralize it?

Comment (nenaddedic) [reply integration/run_test.go]: Oh, OK, I didn't know that.  Then let's leave it.
